File: shkoli/LISH 07 - Varna/work iskren on Pc4-kl1/ioi/robot_sample/robot.sample.pas
program robot;

var
   response      : string;
   width, height : longint;
   
begin
   repeat
      writeln('left');
      flush(output);
      readln(response);
   until response = 'fail';

   width := 0;
   repeat
      width := width+1;
      writeln('right');
      flush(output);
      readln(response);
   until response = 'fail';
   
   repeat
      writeln('up');
      flush(output);
      readln(response);
   until response = 'fail';

   height := 0;
   repeat
      height := height+1;
      writeln('down');
      flush(output);
      readln(response);
   until response = 'fail';
   
   writeln(width*height);
   flush(output);
end.

File: shkoli/LISH 07 - Varna/work iskren on Pc4-kl1/ioi/robot_sample/robot.sample.c
#include <stdio.h>
#include <string.h>

int main()
{
   char response[10];
   int  width, height;

   do {
      printf("left\n");
      fflush(stdout);
      scanf("%s", response);
   } while (strcmp(response, "fail") != 0);

   width = 0;
   do {
      ++width;
      printf("right\n");
      fflush(stdout);
      scanf("%s", response);
   } while (strcmp(response, "fail") != 0);

   do {
      printf("up\n");
      fflush(stdout);
      scanf("%s", response);
   } while (strcmp(response, "fail") != 0);

   height = 0;
   do {
      ++height;
      printf("down\n");
      fflush(stdout);
      scanf("%s", response);
   } while (strcmp(response, "fail") != 0);

   printf("%d\n", width*height);
   fflush(stdout);
   
   return 0;
}
